bigframes/core/blocks.py

```diff
@@ -1923,6 +1929,16 @@ def join(
             raise NotImplementedError(
                 f"Only how='outer','left','right','inner' currently supported. {constants.FEEDBACK_LINK}"
             )
+        # Special case for null index,
+        if (
+            (self.index.nlevels == other.index.nlevels == 0)
+            and (sort is False)
+            and (block_identity_join is False)
```

Comment:
Since these are only bool per type checking, we don't have to worry about any false-y values to mess up "not"

```suggestion
            and not sort
            and not block_identity_join
```

Reply:
Question: Doesn't `block_identity_join` imply indexless join _is_ allowed? What's this checking for?

Edit: found it

https://github.com/googleapis/python-bigquery-dataframes/blob/ca284cc21d5386b4817ecab85b68235456361815/bigframes/core/blocks.py#L2292

Apparently `block_identity_join` is the opposite of what I thought it is. Please add a docstring to this function explaining what these args mean.

Reply:
added docstring